include/cppast/cpp_function.hpp

```diff
@@ -226,6 +226,12 @@ class cpp_function final : public cpp_function_base
         {
             function->constexpr_ = true;
         }
+
+        /// \effects Marks the function as `consteval`.
+        void is_consteval()
+        {
+            function->consteval_ = true;
```

Comment:
This API allows a function that is both constexpr and consteval, please add an assert that this doesn't happen.

Reply:
I meant, please add an assert to `is_consteval()` and `is_constexpr()` to the builder here, not in the libclang code.